### ISSUE TITLE
Inconsistent `$page.state` update when using multiple pushState

### ISSUE THREAD

**Issue by Tyrenn** [sveltejs/kit#11839]
### Describe the bug

When using multiple pushState in a row and then using native ``history.back()`` method or back button, the ``$page.state`` seems to be updated in a weird way :
First time going back, it seems not to update and reactivity isn't fired.
Next times, it seems to be updated twice.

This is pretty annoying as I need, in my application, to be able to push multiple states to update research filters and corresponding query parameters.
I don't know if I'm using this feature wrong or if it hasn't been built for this usecase, but i've foundd myself implementing my own push state handler based on native methods which throw a warning.

### Reproduction

Clicking multiple times on show rectangle and then hide rectangle to generate states using pushState. Then going back while observing the console :

_Clicking show Rectangle_
`> change state {show: true}`
_Clicking hide Rectangle_
`> change state {show: false}`
_Clicking show Rectangle_
`> change state {show: true}`
_Clicking hide Rectangle_
`> change state {show: false}`

_Clicking back history_
`> Popping state`

_Clicking back history_
`> change state {show: true}`
`> change state {show: false}`
`> Popping state`

[https://github.com/Tyrenn/kit_pushstate_issue](https://github.com/Tyrenn/kit_pushstate_issue)

### System Info

```Shell
System:
    OS: Windows 11 10.0.22631
    CPU: (8) x64 11th Gen Intel(R) Core(TM) i7-11370H @ 3.30GHz
    Memory: 4.44 GB / 15.69 GB
  Binaries:
    Node: 18.13.0 - C:\Program Files\nodejs\node.EXE
    npm: 8.1.3 - C:\Program Files\nodejs\npm.CMD
  Browsers:
    Edge: Chromium (121.0.2277.112)
    Internet Explorer: 11.0.22621.1
  npmPackages:
    @sveltejs/adapter-auto: ^3.1.1 => 3.1.1
    @sveltejs/kit: ^2.5.0 => 2.5.0
    @sveltejs/vite-plugin-svelte: ^3.0.2 => 3.0.2
    svelte: ^4.2.10 => 4.2.10
    vite: ^5.0.12 => 5.0.12
```

### Severity

annoyance

**Comment by PatrickG** [comment]
This should be fixed since Sveltekit `2.5.1`.
@Tyrenn can you confirm this?

**Comment by Tyrenn** [comment]
I've just checked and it seems to be working ! Awesome :)
